titiler/api/deps.py

```diff
@@ -230,3 +232,19 @@ def __init__(
         kwargs.pop("bounds", None)
         kwargs.pop("assets", None)  # For STAC
         self.kwargs = kwargs
+
+
+class CommonMosaicParams:
+    """Common mosaic params."""
+
+    def __init__(
+        self, url: str = Query(..., description="MosaicJSON URL"),
+    ):
+        """Create mosaic path from args"""
+        parsed = urlparse(url)
+        if parsed.scheme == "mosaicid":
+            self.mosaic_path = (
+                f"{DEFAULT_MOSAIC_BACKEND}{DEFAULT_MOSAIC_HOST}/{parsed.netloc}.json.gz"
+            )
+        else:
+            self.mosaic_path = url
```

Comment:
Looking at this, I don't think we need to use a class here, a simple function that returns a `mosaic_path` should be sufficient ?